### PR TITLE
Feature/#41 TruckCategoryService

### PR DESCRIPTION
**Pull request by botirjonabdumannonovv**
- Created TruckCategoryService
 - Created CRUD operations in TruckCategoryService
 - Created DataContex

### CODE REVIEW

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq.Expressions;
+
+using Training.TruckWorld.Backend.Domain.Entities;
+
+namespace Training.TruckWorld.Backend.Application.Trucks.Models.Interface;
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs]: Trucks deb ochilgan folderga services deb folder ochib uni ichiga shu interface ni ko'chiring

Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs]: keyin hech qanaqa interfaces deb folder ochib o'tirmang

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs:1]: hammasi zo'r faqat folderlarida adashibsiz salgina

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs (hunk 1 above)
Comment (botirjonabdumannonovv) [reply src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs]: hop

Comment (botirjonabdumannonovv) [reply src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs]: tushunarli

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs
Comment (botirjonabdumannonovv) [reply src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs:1]: motivatsiya

**Comment by botirjonabdumannonovv** [comment]
- Edited folder location

**Review by azizbek-abdurahmonov — DISMISSED** [review]

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs (hunk 1 above)
Comment (Habiba67y) [reply src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs]: 👍

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs
Comment (Habiba67y) [reply src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Models/Services/ITruckCategoryService.cs:1]: 😁

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Services/ITruckCategoryService.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq.Expressions;
+
+using Training.TruckWorld.Backend.Domain.Entities;
+
+namespace Training.TruckWorld.Backend.Application.Trucks.Models.Interface;
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Services/ITruckCategoryService.cs]: folderlarni o'zgartirgan bo'sez namespace larni ham to'g'irlab qo'yasiz

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Trucks/Services/TruckCategoryService.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
+﻿using System.Linq.Expressions;
+using Training.TruckWorld.Backend.Application.Trucks.Services;
+using Training.TruckWorld.Backend.Domain.Entities;
+using Training.TruckWorld.Backend.Persistence.DataContexts;
+
+namespace Training.TruckWorld.Backend.Infrastructure.Trucks.Services;
```
Comment (Habiba67y) [inline src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Infrastructure/Trucks/Services/TruckCategoryService.cs:6]: bunda ham

**Review by Habiba67y — APPROVED** [review]

**Review by azizbek-abdurahmonov — APPROVED** [review]

File: src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Services/ITruckCategoryService.cs (hunk 2 above)
Comment (botirjonabdumannonovv) [reply src/Training.TruckWorld.Clone.Backend/Training.TruckWorld.Backend/Application/Trucks/Services/ITruckCategoryService.cs]: namespace edited

**Comment by botirjonabdumannonovv** [comment]
- namespace edited

**Pull request merged**